### PR TITLE
[docs] Update docs for building from source

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

The existing docs didn't work for me and these updates did. 🤷‍♀️ I selectively pulled this stuff out of the CI (which ideally would just be runnable locally). 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-contribute/development.rst [hunk 1]
```diff
@@ -143,11 +143,20 @@ To build Ray on Ubuntu, run the following commands:
 .. code-block:: bash
 
   sudo apt-get update
-  sudo apt-get install -y build-essential curl unzip psmisc
+  sudo apt-get install -y build-essential curl gcc-9 g++-9 pkg-config psmisc unzip
+  sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90 \
```
[the remark this answers is not in the document]
Comment (scv119) [reply doc/source/ray-contribute/development.rst:147]: on ubuntu 18.04, you need add 
```
sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
```
 before `sudo apt-get update`.

**Review by scv119 — CHANGES_REQUESTED** [review]
need support 18.04

**Comment by scv119** [comment]
friendly ping!

**Comment by wuisawesome** [comment]
I've pushed your comment but haven't had time to spin up 18.04 vm for manual testing (i'm using 22.04 these days)

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
Thanks for the fix!

**Review by richardliaw — APPROVED** [review]
stamp

**Pull request merged**